TypeSystem/Reflection/types.h

```diff
@@ -127,8 +129,11 @@ inline uint64_t ROL64(const TypeID type_id, size_t nbits) {
   return current::ROL64(static_cast<uint64_t>(type_id), nbits);
 }
 
-inline TypeID CalculateTypeID(const ReflectedType_Struct& s) {
+inline TypeID CalculateTypeID(const ReflectedType_Struct& s, bool is_incomplete = false) {
   uint64_t hash = bricks::CRC32(s.name);
+  if (is_incomplete) {
+    return static_cast<TypeID>(TYPEID_INCOMPLETE_TYPE + hash);
+  }
```

Comment:
Still suspicious about `i + 8` and `i + 16` below. Can they be some `i + 19` and `i + 29` instead? :)